api/views/course_admin.py

```diff
@@ -40,6 +43,28 @@ def register_user(self, request, pk=None):
         if course.is_registered(user):
             raise ValidationError(ERROR_MESSAGES.COURSE_REGISTRATION.ALREADY_REGISTERED)
 
-        # TODO: Implement the registration
+        # TODO: Implement the registration that uses of canvas_user_id which can be done from canvas api
 
         return Response(request.data)
+
+    @action(detail=True, methods=['post'], url_path='change-status')
+    def change_status(self, request, pk=None):
+        '''
+        Action that will update the block/verify status of a canvascourseregistration object.
+        '''
+        registration_id = request.data.get("id")
+        status = request.data.get("status")
+        status_type = request.data.get("type")
+        course_registration = get_object_or_404(CanvasCourseRegistration, id=registration_id)
+
+        if status_type == 1:
+            course_registration.is_blocked = status
+            course_registration.save()
+
+        if status_type == 2:
+            course_registration.is_verified = status
+            course_registration.save()
```

Comment:
What exactly is `status_type` here? If it is arbitrary, then maybe make it descriptive, instead of just numbers which aren't readable

Reply:
I changed that. Is it appropriate now?